src/subtle/batched-effect.ts

```diff
@@ -0,0 +1,49 @@
+import { Signal } from "signal-polyfill";
+
+const notifiedEffects = new Set<{
+  computed: Signal.Computed<void>;
+  watcher: Signal.subtle.Watcher;
+}>();
+
+/**
+ * Runs a function inside a batch, and calls all the effected batched effects
+ * synchronously.
+ */
+export const batch = (fn: () => void) => {
```

Comment:
why const arrow instead of `function batch`?
(is this personal preference, or some perf/debugging thing I don't know about? :sweat_smile: )

Reply:
Personal preference - I just use arrow functions for everything possible.

I supposed there's a _slight_ theoretical perf benefit from not having a mutable binding, if the VM correctly elides the TDZ check for this case. Maybe some bundlers take advantage of this too? There's also of course a theoretical perf hit from un-elided TDZ checks.

I just personally think that arrow functions are the correct function semantics, and use the `function` to highlight cases where `this` might be non-lexical.